File: 3612-adjacent-increasing-subarrays-detection-i/3612-adjacent-increasing-subarrays-detection-i.cpp
class Solution {
public:
    bool isstrictlyincreasing(int i,int k,vector<int>&nums){
        int n=nums.size();
        for(int j=i;j<i+k-1;j++){
            if(nums[j]>=nums[j+1])
            return false;
        }
        return true;
    }
    bool hasIncreasingSubarrays(vector<int>& nums, int k) {
        int n=nums.size();
        if(n<2*k) return false;
        for(int i=0;i<n-(2*k)+1;i++){
            if(isstrictlyincreasing(i,k,nums)&&isstrictlyincreasing(i+k,k,nums))
            return true;
           
        }
        return false;
        
    }
};

File: 3612-adjacent-increasing-subarrays-detection-i/3612-adjacent-increasing-subarrays-detection-i.py
class Solution(object):

    def hasIncreasingSubarrays(self, nums, k):
        if len(nums)<2*k:
            return False
        def isincreasing(nums,s,k):
            for i in range(s,s+k-1):
                if(nums[i]>=nums[i+1]):
                    return False
            return True
        l=len(nums)-(2*k)+1
        for i in range(0,l):
            if(isincreasing(nums,i,k)and isincreasing(nums,i+k,k)):
                return True
        return False

        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        